contract/models/abstract_contract_line.py

```diff
@@ -125,9 +125,27 @@ class ContractAbstractContractLine(models.AbstractModel):
         required=True,
         ondelete='cascade',
     )
-    display_type = fields.Selection([
-        ('line_section', "Section"),
-        ('line_note', "Note")], default=False, help="Technical field for UX purpose.")
+    display_type = fields.Selection(
+        selection=[
+            ('line_section', "Section"),
+            ('line_note', "Note"),
+        ],
+        default=False,
+        help="Technical field for UX purpose."
+    )
+    notes_invoicing_mode = fields.Selection(
+        selection=[
+            ('with_previous_line', 'With previous line'),
+            ('with_next_line', 'With next line'),
+            ('custom', 'Custom'),
+        ],
+        default='with_previous_line',
+        help="Defines when the Note is invoiced:\n"
+             "- With previous line: If the previous line can be invoiced.\n"
+             "- With next line: If the previous line can be invoiced.\n"
+             "- Custom: Depending on the recurrence to be define."
+    )
+    is_recurring_note = fields.Boolean(compute="_compute_is_recurring_note")
```

Comment:
No need of this extra field as the visibility computation can be done with domain and it's very simple